infra/testApi.mjs

```diff
@@ -1,82 +1,163 @@
 import { strict as assert } from 'node:assert';  // use strict equality everywhere 
 
-// Future TODO: before this API becomes set in stone/offered publically, we should change the results of these methods to be just the output data rather than duplicating input values.
+// Future TODO: before this API becomes set in stone/offered publicly, we should change the results of these methods to be just the output data rather than duplicating input values.
 
-const sample = (await(await fetch('http://127.0.0.1:8000/api/sample', { method: 'POST', body: JSON.stringify({ formula: '(FPCore (x) (- (sqrt (+ x 1))))', seed: 5 }) })).json())
+// Reusable testing data
+const FPCoreFormula = '(FPCore (x) (- (sqrt (+ x 1)) (sqrt x)))'
+const FPCoreFormula2 = '(FPCore (x) (- (sqrt (+ x 1))))'
+const eval_sample = [[[1], -1.4142135623730951]]
+
+// improve endpoint
+const improveURL = `http://127.0.0.1:8000/improve?formula=${encodeURIComponent(FPCoreFormula2)}`
+const improveResponse = await fetch(improveURL, { method: 'GET' })
+assert.equal(improveResponse.status, 200)
+// This test is a little flaky as the character count of the response is not consistent.
+// const improveHTML = await improveResponse.text()
+// const improveHTMLexpectedCount = 25871
+// assert.equal(improveHTML.length, improveHTMLexpectedCount, `HTML response character count should be ${improveHTMLexpectedCount} unless HTML changes.`)
+
+// improve-start endpoint
+const startURL = `http://127.0.0.1:8000/improve-start?formula=${encodeURIComponent(FPCoreFormula2)}`
+const startResponse = await fetch(startURL, { method: 'POST' })
```

Comment:
Probably me not understanding out to use the API and stumbled into this working. Though changing this to GET responds with a 404. 

This is why I was asking about the API calling order. the improve endpoints feel very detached from the rest of them. What's the cast for a `put` request with Herbie in regards to the `improve` endpoint?

Reply:
We should not use PUT. I think we should try to pass the `formula` in a JSON object like the rest of the API, instead of as a query parameter. This will require changes to `demo.js` as well as the backend & the tests.